### PR TITLE
feat(builder): add having filter

### LINKED ISSUE #2309: FE(QB): Add Having Filter
Screenshot 2023-02-22 at 9 26 28 AM

The having filter will always work on the aggregation data i.e it will show the option when there is some aggregation performed.

Say the user does `MAX` (selected aggregate operation of the query) of `span.credit_limit` (selected aggregate attribute of the query); then we need to show all the options for this i.e

`MAX(span.credit_limit)` < (the user will enter this input. there is no API call involved here)
`MAX(span.credit_limit)` > (...)
`MAX(span.credit_limit)` = (...)
`MAX(span.credit_limit)` != (...) 
...

This is not a signal-specific component functionality. This can be done once and used everywhere.

### PR DESCRIPTION
**Pull request by yeshev**
close #2309

### CODE REVIEW

**Review by palashgdev — DISMISSED** [review]

File: frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+					num.toLowerCase().includes(op.toLowerCase()),
+				);
+
+				newOptions = filteredOperators.map((opt) => ({
+					label: `${columnName} ${opt} ${value}`,
+					value: `${columnName} ${opt} ${value}`,
+				}));
+			}
+
+			setOptions(newOptions);
```
Comment (palashgdev) [inline frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx:85]: can we make a helper function for this one

File: frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { Having } from 'types/api/queryBuilder/queryBuilderData';
+import { SelectOption } from 'types/common/select';
+
+// ** Types
+import { HavingFilterProps } from './HavingFilter.interfaces';
+
+const { Option } = Select;
+
+const initialFormValues: Having = {
```
Comment (palashgdev) [inline frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx:19]: can we put this one in config

**Review by palashgdev — CHANGES_REQUESTED** [review]
small comments apart form that LGTM

File: frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx (hunk 1 above)
Comment (yeshev) [reply frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx:85]: This function has a specific case for generating options, so i don't think it makes sense to create a separated function for it and use only in one place.

Comment (palashgdev) [reply frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx:82]: i know we are using in one place only but how you would write unit test case for same ?

Comment (yeshev) [reply frontend/src/container/QueryBuilder/filters/HavingFilter/HavingFilter.tsx:82]: I've already written unit test for this component and i wrote the expected result when we generate options after click. For component we need to check if the render is correct.

**Comment by srikanthccv** [comment]
@palashgdev, this looks good from the functionality perspective.

**Review by palashgdev — APPROVED** [review]

**Pull request merged**